librato.go

```diff
@@ -21,18 +22,40 @@ func translateTimerAttributes(d time.Duration) (attrs map[string]interface{}) {
 	return
 }
 
+type RateValue string
+
+const (
+	Rate1  RateValue = "Rate1"
+	Rate5  RateValue = "Rate5"
+	Rate15 RateValue = "Rate15"
+)
+
+type rateFn func(r RateValue) interface{}
+
+//Can add other options to not conform to default behavior
+//Doing this allows us to reduce the number of metric streams emitted by default
+type EnableRateSet struct {
+	Timer map[RateValue]bool
+	Meter map[RateValue]bool
+}
+
 type Reporter struct {
 	Email, Token    string
 	Source          string
 	Interval        time.Duration
 	Registry        metrics.Registry
 	Percentiles     []float64              // percentiles to report on histogram metrics
 	TimerAttributes map[string]interface{} // units in which timers will be displayed
+	RateAttributes  *EnableRateSet
 	intervalSec     int64
 }
 
 func NewReporter(r metrics.Registry, d time.Duration, e string, t string, s string, p []float64, u time.Duration) *Reporter {
-	return &Reporter{e, t, s, d, r, p, translateTimerAttributes(u), int64(d / time.Second)}
+	return &Reporter{e, t, s, d, r, p, translateTimerAttributes(u), nil, int64(d / time.Second)}
+}
+
+func NewReporterWithRateOptions(r metrics.Registry, o EnableRateSet, d time.Duration, e string, t string, s string, p []float64, u time.Duration) *Reporter {
```

Comment:
I personally prefer the pattern of adding one method of the following form per one optional member. Otherwise, you end up with a combinatoric explosion of constructors every time you add a new optional thing: 

```go
func (self *Reporter) WithXType(someX XType) {
    self.x = someX; return self
}
```

If you can switch that up, I'll happily merge, rest looks great